components/base/wheeled/kinematics.go

```diff
@@ -23,13 +21,25 @@ type kinematicWheeledBase struct {
 
 // WrapWithKinematics takes a wheeledBase component and adds a slam service to it
 // It also adds kinematic model so that it can be controlled.
-func WrapWithKinematics(ctx context.Context, base *wheeledBase, slam slam.Service) (base.KinematicBase, error) {
-	var err error
+func (base *wheeledBase) WrapWithKinematics(ctx context.Context, slamSvc slam.Service) (base.KinematicBase, error) {
 	kwb := &kinematicWheeledBase{
 		wheeledBase: base,
-		slam:        slam,
+		slam:        slamSvc,
 	}
-	kwb.model, err = kwb.buildModel(ctx)
+	// gets the extents of the SLAM map
+	data, err := slam.GetPointCloudMapFull(ctx, slamSvc)
+	if err != nil {
+		return nil, err
+	}
+	dims, err := pointcloud.GetPCDMetaData(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+	kwb.model, err = Model(
+		kwb.name,
+		kwb.collisionGeometry,
+		[]referenceframe.Limit{{Min: dims.MinX, Max: dims.MaxX}, {Min: dims.MinZ, Max: dims.MaxZ}},
```

Comment:
I think this is a product of cartographer but I'm not entirely sure.

Reply:
What are min/max Y on these maps? Both zero?

Since we mostly are only using cartographer at the moment, I'm fine with leaving this generally, but would like if we add an error if either +/- Z limits are both 0, or if any Y limit is *not* zero

Reply:
2D (carto) will keep Y the same along the map it has built, and XZ forms the plane of the map. This can obviously change with 3D slam. 

I don't think it is generally enforced. @nicksanford? Long therm there needs to be a transform sent in from the lidar and/or algorithm. Or this is just the contract that we formalize, but that seems tricky to impose.  Maybe a comment/todo is appropriate here.

Reply:
My understanding (which may well be incomplete) is that right now there is no contract specifying which dimensions non-viam-carto 2d slam will use.